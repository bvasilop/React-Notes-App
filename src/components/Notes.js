import React from 'react';
import PropTypes from 'prop-types';

const Notes = ({ error, note, removeNote }) => (
  <div>
    {note.title.length === 0 && (
      <div className="ui pointing below prompt label">
        Please enter title below
      </div>
    )}
    <h2 className="item">{note.title}</h2>
    <ul>
      <li>
        {note.body.length === 0 && (
          <div className="ui pointing below prompt label">
            Please enter note contents below
          </div>
        )}
        <p>{note.body}</p>
      </li>
    </ul>
    <button
      type="submit"
      className="ui button"
      onClick={() => removeNote(note.title)}
    >
      remove
    </button>
  </div>
);

Notes.propTypes = {
  error: PropTypes.object,
  note: PropTypes.object,
  removeNote: PropTypes.func,
};
export default Notes;
