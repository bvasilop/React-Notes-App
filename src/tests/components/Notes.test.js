import React from 'react';
import { shallow } from 'enzyme';
import Notes from '../../components/Notes';

const title = '';
const body = '';

test('should render NoteApp correctly', () => {
  const wrapper = shallow(<Notes notes={(title, body)} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render h1 correctly', () => {
  const wrapper = shallow(<Notes notes={(title, body)} />);
  expect(wrapper.find('h1').length).toBe(1);
  expect(wrapper.find('h1').text()).toBe('Notes');
  expect(wrapper).toMatchSnapshot();
});

test('should render p correctly', () => {
  const wrapper = shallow(<Notes notes={(title, body)} />);
  expect(wrapper.find('p').length).toBe(1);
  // expect(wrapper.find('p').text()).toBe('');
  expect(wrapper).toMatchSnapshot();
});

test('should render button correctly', () => {
  const wrapper = shallow(<Notes notes={(title, body)} />);
  expect(wrapper.find('button').length).toBe(1);
  // expect(wrapper.find('button').text()).toBe('remove');
  expect(wrapper).toMatchSnapshot();
});

test('should render error for invalid submission', () => {
  const wrapper = shallow(<Notes />);
  expect(wrapper).toMatchSnapshot();
  wrapper.find('button').simulate('submit', {
    preventDefault: () => {},
  });
});
