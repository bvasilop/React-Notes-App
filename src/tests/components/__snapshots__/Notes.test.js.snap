// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NoteApp correctly 1`] = `
<div
  className="ui raised very padded text container segment"
>
  <span
    className="ui center aligned icon header"
  >
    <i
      className="pencil alternate icon"
    />
    <p>
      React Note App
    </p>
  </span>
  <h1
    className="ui center aligned header"
  >
    Notes
  </h1>
  <div
    className="ui segment"
  >
    <strong>
      Add some notes!
    </strong>
  </div>
  <div
    className="ui segment"
  >
    <h2
      className="ui center aligned header"
    >
      Add Note
    </h2>
    <form
      className="ui form"
      onSubmit={[Function]}
    >
      <label
        htmlFor="Title"
      >
        Title
        <input
          className="ui input focus"
          name="title"
          onChange={[Function]}
          placeholder="enter title of note"
          type="text"
          value=""
        />
      </label>
      <label
        htmlFor="Body"
      >
        Body
        <textarea
          className="ui fluid action input"
          onChange={[Function]}
          placeholder="enter contents of note"
          value=""
        />
      </label>
      <button
        className="ui button"
        type="submit"
      >
        add note
      </button>
    </form>
  </div>
</div>
`;

exports[`should render button correctly 1`] = `
<div
  className="ui raised very padded text container segment"
>
  <span
    className="ui center aligned icon header"
  >
    <i
      className="pencil alternate icon"
    />
    <p>
      React Note App
    </p>
  </span>
  <h1
    className="ui center aligned header"
  >
    Notes
  </h1>
  <div
    className="ui segment"
  >
    <strong>
      Add some notes!
    </strong>
  </div>
  <div
    className="ui segment"
  >
    <h2
      className="ui center aligned header"
    >
      Add Note
    </h2>
    <form
      className="ui form"
      onSubmit={[Function]}
    >
      <label
        htmlFor="Title"
      >
        Title
        <input
          className="ui input focus"
          name="title"
          onChange={[Function]}
          placeholder="enter title of note"
          type="text"
          value=""
        />
      </label>
      <label
        htmlFor="Body"
      >
        Body
        <textarea
          className="ui fluid action input"
          onChange={[Function]}
          placeholder="enter contents of note"
          value=""
        />
      </label>
      <button
        className="ui button"
        type="submit"
      >
        add note
      </button>
    </form>
  </div>
</div>
`;

exports[`should render error for invalid submission 1`] = `
<div
  className="ui raised very padded text container segment"
>
  <span
    className="ui center aligned icon header"
  >
    <i
      className="pencil alternate icon"
    />
    <p>
      React Note App
    </p>
  </span>
  <h1
    className="ui center aligned header"
  >
    Notes
  </h1>
  <div
    className="ui segment"
  >
    <strong>
      Add some notes!
    </strong>
  </div>
  <div
    className="ui segment"
  >
    <h2
      className="ui center aligned header"
    >
      Add Note
    </h2>
    <form
      className="ui form"
      onSubmit={[Function]}
    >
      <label
        htmlFor="Title"
      >
        Title
        <input
          className="ui input focus"
          name="title"
          onChange={[Function]}
          placeholder="enter title of note"
          type="text"
          value=""
        />
      </label>
      <label
        htmlFor="Body"
      >
        Body
        <textarea
          className="ui fluid action input"
          onChange={[Function]}
          placeholder="enter contents of note"
          value=""
        />
      </label>
      <button
        className="ui button"
        type="submit"
      >
        add note
      </button>
    </form>
  </div>
</div>
`;

exports[`should render h1 correctly 1`] = `
<div
  className="ui raised very padded text container segment"
>
  <span
    className="ui center aligned icon header"
  >
    <i
      className="pencil alternate icon"
    />
    <p>
      React Note App
    </p>
  </span>
  <h1
    className="ui center aligned header"
  >
    Notes
  </h1>
  <div
    className="ui segment"
  >
    <strong>
      Add some notes!
    </strong>
  </div>
  <div
    className="ui segment"
  >
    <h2
      className="ui center aligned header"
    >
      Add Note
    </h2>
    <form
      className="ui form"
      onSubmit={[Function]}
    >
      <label
        htmlFor="Title"
      >
        Title
        <input
          className="ui input focus"
          name="title"
          onChange={[Function]}
          placeholder="enter title of note"
          type="text"
          value=""
        />
      </label>
      <label
        htmlFor="Body"
      >
        Body
        <textarea
          className="ui fluid action input"
          onChange={[Function]}
          placeholder="enter contents of note"
          value=""
        />
      </label>
      <button
        className="ui button"
        type="submit"
      >
        add note
      </button>
    </form>
  </div>
</div>
`;

exports[`should render p correctly 1`] = `
<div
  className="ui raised very padded text container segment"
>
  <span
    className="ui center aligned icon header"
  >
    <i
      className="pencil alternate icon"
    />
    <p>
      React Note App
    </p>
  </span>
  <h1
    className="ui center aligned header"
  >
    Notes
  </h1>
  <div
    className="ui segment"
  >
    <strong>
      Add some notes!
    </strong>
  </div>
  <div
    className="ui segment"
  >
    <h2
      className="ui center aligned header"
    >
      Add Note
    </h2>
    <form
      className="ui form"
      onSubmit={[Function]}
    >
      <label
        htmlFor="Title"
      >
        Title
        <input
          className="ui input focus"
          name="title"
          onChange={[Function]}
          placeholder="enter title of note"
          type="text"
          value=""
        />
      </label>
      <label
        htmlFor="Body"
      >
        Body
        <textarea
          className="ui fluid action input"
          onChange={[Function]}
          placeholder="enter contents of note"
          value=""
        />
      </label>
      <button
        className="ui button"
        type="submit"
      >
        add note
      </button>
    </form>
  </div>
</div>
`;
